backend.go

```diff
@@ -128,6 +133,54 @@ func (r *RedisBackend) IncrementID(ctx context.Context, id int) (int64, error) {
 	return cnt, nil
 }
 
+func (r *RedisBackend) GetMessageReply(ctx context.Context, msg MessageIdentifier) ([]Message, error) {
+	log.Info().Str("return_queue", msg.Retqueue).Msg("Waiting reply")
+	responses := []Message{}
+	retQueueLen, err := r.client.LLen(ctx, msg.Retqueue).Result()
+	if err != nil {
+		log.Error().Err(err).Msg("error fetching from redis")
+		return responses, err
+	}
+	// TODO: check if the message ready
+	if int(retQueueLen) < 1 {
+		return nil, errors.New("The message not handled yet, it is in progress.")
```

Comment:
it's not necessarily in progress, the request might have a wrong retqueue. Better if it's ErrNotAvailable to make it easy for the caller to differentiate between the error when the message isn't and when redis is not available for example.